Add PHP version check for plugin.

rahulsprajapati opened the pull request:
<!--
Filling out this template is required.  Any PR that does not include enough information to be reviewed may be closed at a maintainers' discretion.  All new code requires documentation and tests to ensure against regressions.
-->

### Description of the Change
<!--
We must be able to understand the design of your change from this description.  The maintainer reviewing this PR may not have worked with this code recently, so please provide as much detail as possible.

Where possible, please also include:
- verification steps to ensure your change has the desired effects and has not introduced any regressions
- any benefits that will be realized
- any alternative implementations or possible drawbacks that you considered
- screenshots or screencasts
-->
Add minimum version check for the plugin before loading it. This will ensure plugin update doesn't break the site that don't match our minimum PHP version.

<img width="1039" alt="image" src="https://github.com/10up/maps-block-apple/assets/10358350/cea3e319-19e5-47ab-a221-35b2398262dc">


<!-- Enter any applicable Issue number(s) here that will be closed/resolved by this PR. -->
Closes #172

### How to test the Change
<!-- Please provide steps on how to test or validate that the change in this PR works as described. -->

1. Setup a WordPress environment running PHP 7.3
2. Install Block for Apple Maps plugin by downloading it with this [feature](https://github.com/10up/maps-block-apple/archive/refs/heads/fix/172-php-ver-check.zip) branch.
3. Verify that plugin functionality doesn't load and you see an admin error message for php version. ( refer above screenshot in Description )

### Changelog Entry
<!--
Please include a summary for this PR, noting whether this is something being Added / Changed / Deprecated / Removed / Fixed / or Security related.  You can replace the sample entries after this comment block with the single changelog entry line for this PR. -->
> Fixed - Better error handling for environments that don't match our minimum PHP version

### Credits
<!-- Please list any and all contributors on this PR so that they can be added to this projects CREDITS.md file. -->
Props @dkotter, @rahulsprajapati 


### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you are unsure about any of these, please ask for clarification.  We are here to help! -->
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

rahulsprajapati committed:
Add PHP version check for plugin.

dkotter commented on maps-block-apple.php:
```diff
@@ -3,7 +3,7 @@
  * Plugin URI:        https://github.com/10up/maps-block-apple
  * Description:       An Apple Maps block for the WordPress block editor (Gutenberg).
- * Version:           1.1.1
+ * Version:           1.1.2
```
Version numbers will be bumped when we do a release so no need to update those in this PR

rahulsprajapati committed:
Revert version update change.

rahulsprajapati committed:
Remove duplicate php version check.
- Activation is checking 5.8 as min version, but as per plugin header
  php 7.4 is min requirement.

dkotter commented on maps-block-apple.php:
```diff
@@ -25,20 +25,47 @@
+ *
+ * @return string Minimum version required.
  */
-register_activation_hook(
```
Do we want to get rid of this check? This seems different from the PHP version check we are adding here

rahulsprajapati replied:
We are not loading the plugin functionality if minimum php version requirement does not match, and we show the admin notice for the same. 
I thought this check on activation looked redundant, Since we will still see the admin notice / error on plugin activation if php version didn't match the requirement, and plugin won't load.

dkotter replied:
So the check we're removing isn't looking at the PHP version, it's looking at the WordPress version (it shows an error if using WordPress <5.8. I'm not sure that check is needed but also want to ensure we're not getting rid of it if it is needed. Thinking we may want to add a WordPress `version_compare` within our new conditional so we keep that check. Something like:

```php
if ( ! site_meets_php_requirements() || ! version_compare( $GLOBALS['wp_version'], '5.8', '>=' ) ) {
```

dkotter replied:
Or another approach is how we do this in Distributor: https://github.com/10up/distributor/blob/develop/distributor.php#L93

jeffpaul commented:
@ravinderk as you're able during your OSS time, if you could please help handle the code review items here that would be great... thanks!

ravinderk committed:
chore: install environmet validation composer packgae

ravinderk committed:
feat: use 10up wp-compat-validation-toll lib for php validation

ravinderk committed:
dev: update build release zip workflow

insrall composer packages.

ravinderk committed:
doc: add file description

ravinderk committed:
format: improve code formatting

ravinderk committed:
format: improve code formatting

ravinderk committed:
format: improve code formatting

ravinderk committed:
Merge branch 'develop' into fix/172-php-ver-check

ravinderk commented:
> Details

@jeffpaul I updated pull request. I found that E2E test are falling. 
@dkotter Do you have any suggestion?

ravinderk commented:
> @ravinderk as you're able during your OSS time, if you could please help handle the code review items here that would be great... thanks!

@jeffpaul I updated pull request. I found that E2E test are falling. 
@dkotter Do you have any suggestion?

dkotter commented:
> @dkotter Do you have any suggestion?

We need to add `composer install --no-dev` to our `cypress.yml` file, otherwise our dependency check fails and the settings never get added (and thus we get a 403 error in our tests when trying to access that settings page)

ravinderk committed:
dev: install composer for WordPress release workflow

ravinderk committed:
dev: use latest E2E workflow config to check E2E tests

ravinderk commented:
> > @dkotter Do you have any suggestion?
> 
> We need to add `composer install --no-dev` to our `cypress.yml` file, otherwise our dependency check fails and the settings never get added (and thus we get a 403 error in our tests when trying to access that settings page)

@dkotter I made the change to `build-release-zip.yml` in this branch, but `cypress.yml` is using `build-release-zip.yml@develop` to create a build for testing.  I temporarily replaced the build workflow with `build-release-zip.yml@fix/172-php-ver-check` to validate E2E tests. I will revert back to the changes before merging the pull request.

dkotter commented on .github/workflows/cypress.yml:
```diff
@@ -5,8 +5,8 @@ on:
-  build: 
-    uses: 10up/maps-block-apple/.github/workflows/build-release-zip.yml@develop
+  build:
+    uses: 10up/maps-block-apple/.github/workflows/build-release-zip.yml@fix/172-php-ver-check
```
Making a note that this needs updated before merge

dkotter commented on includes/environment-validation.php:
```diff
@@ -0,0 +1,29 @@
+ * This file is used to validate the environment.
+ *
+ * @package tenup\Maps_Block_Apple
+ * @since   x.x.x
```
Making a note that this should be updated before merge

ravinderk committed:
chore: cleanup
